app/models/document_set.rb

```diff
@@ -27,121 +27,107 @@ class DocumentSet < ApplicationRecord
   include DocumentSetStatistic
 
   extend FriendlyId
-  friendly_id :slug_candidates, :use => [:slugged, :history]
+  friendly_id :slug_candidates, use: [:slugged, :history]
   before_save :uniquify_slug
   # validate :slug_uniqueness_across_objects
 
   mount_uploader :picture, PictureUploader
 
-  belongs_to :owner, :class_name => 'User', :foreign_key => 'owner_user_id', optional: true
+  belongs_to :owner, class_name: 'User', foreign_key: 'owner_user_id', optional: true
   belongs_to :collection, optional: true
-  belongs_to :next_untranscribed_page, foreign_key: 'next_untranscribed_page_id', class_name: "Page", optional: true
-
-  has_many :pages, through: :works
+  belongs_to :next_untranscribed_page, foreign_key: 'next_untranscribed_page_id', class_name: 'Page', optional: true
 
   has_many :document_set_works
-  has_many :works, -> { order 'title' }, through: :document_set_works
+  has_many :works, -> { order(:title) }, through: :document_set_works
+  has_many :pages, through: :works
+  has_many :articles, -> { distinct }, through: :pages
+  has_many :notes, -> { order(created_at: :desc) }, through: :works
+  has_many :deeds, -> (document_set) {
+    where(work_id: document_set.works.select(:id))
+      .includes(:work)
+      .reorder('deeds.created_at DESC')
+  }, through: :collection, source: :deeds
```

Comment:
We have all the necessary reference_ids/keys here. We can make `articles` and `deeds` an actual relationship rather than a function.

This allows for the possibility of even preloading these relationships which can help with performance.

Reply:
That's a good point.  One question: this code was written before we started experiencing performance problems with `deeds`.  Do you think that that's a worry in the current code?  Can we avoid it in a new relation?

Reply:
I think this is a problem that went undetected, and as your db gets larger so does performance was affected.

The reason I suggest moving this in a relation is to preload it. Currently we have n+1 problems here whenever we call `deeds` for this model.

Reply:
Let's do that, then.